api/orchestrationI_instanceprofiles.go

```diff
@@ -0,0 +1,67 @@
+package api
+
+import (
+	"context"
+
+	"github.com/YaleSpinup/ec2-api/common"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+)
+
+// deleteInstanceProfile deletes the specified instance profile and associated role, if they exist
+// any policies attached to the role will be detached and left intact
+func (o *iamOrchestrator) deleteInstanceProfile(ctx context.Context, name string) error {
+	ip, err := o.iamClient.GetInstanceProfile(ctx, &iam.GetInstanceProfileInput{InstanceProfileName: aws.String(name)})
+	if err != nil {
+		return common.ErrCode("failed to get instance profiles", err)
+	}
+	// detach policies from role(s) and delete the role(s)
+	for _, managedRole := range ip.Roles {
+		roleName := aws.StringValue(managedRole.RoleName)
+		// detach all managed policies
+		rps, err := o.iamClient.ListAttachedRolePolicies(ctx, &iam.ListAttachedRolePoliciesInput{RoleName: aws.String(roleName)})
+		if err != nil {
+			return common.ErrCode("failed to list managed roles", err)
```

Comment:
@tenyo incase of error while processing a role, should we return the error or log the error similar to what we have done in delete image API?
I have same question for other returns in this function.

Reply:
Yes, I think we should return errors for all of these